pkg/module/sync/runner.go

```diff
@@ -110,23 +110,37 @@ func (r *Runner) updateManifests(ctx context.Context, kyma *v1beta2.Kyma,
 		return commonerrs.ErrTypeAssert
 	}
 
-	if err := r.doUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled,
-		manifestObj); err != nil {
+	if err := r.DoUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled, manifestObj); err != nil {
 		return err
 	}
 	module.Manifest = manifestObj
 	return nil
 }
 
-func (r *Runner) doUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
+func (r *Runner) DoUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
 	manifestObj *v1beta2.Manifest,
 ) error {
+	manifestInCluster := &v1beta2.Manifest{}
+	var err error
+	if err = r.Get(ctx, client.ObjectKey{Namespace: manifestObj.GetNamespace(), Name: manifestObj.GetName()},
+		manifestInCluster); err != nil && !util.IsNotFound(err) {
+		return fmt.Errorf("error get manifest %s: %w", client.ObjectKeyFromObject(manifestObj), err)
+	}
+
+	if !NeedToUpdate(manifestInCluster, manifestObj) {
```

Comment:
as mentioned in the original [solution](https://github.com/kyma-project/lifecycle-manager/issues/1456), the version and module channel should coming from `kyma.status`, but not from actual `manifestInCluster`